binaries/cli/src/main.rs

```diff
@@ -466,6 +498,8 @@ fn query_running_dataflows(
     Ok(ids)
 }
 
-fn connect_to_coordinator() -> std::io::Result<Box<TcpRequestReplyConnection>> {
-    TcpLayer::new().connect(control_socket_addr())
+fn connect_to_coordinator(
+    coordinator_addr: SocketAddr,
```

Comment:
How about we take an `Option<SocketAddr>` here? Then we only need to do the `.unwrap_or_else(|| control_socket_addr())` in one place.

Reply:
Yeah,  I will fix it!

Reply:
It seems like that `control_socket_addr` was hard-code and this one is a inner port to handle `ControlRequest`. 

`control_events` function bind this inner port. The actual things that `connect_to_coordinator` has done is  connect daemon to this inner port, then converts every request to an inner event. These are all in `control_events`. 

When you make these changes, you also need to change the code in `control_events` to export the inner port.

Reply:
I think you are right. since the control_socket_addr() is localhost, so it cannot be found with public IP.

```
pub fn control_socket_addr() -> SocketAddr {
    SocketAddr::new(Ipv4Addr::new(127, 0, 0, 1).into(), 6012)
}
```

Reply:
I am guessing that even you make the port public, `dora start dataflow.yaml --coordinator_addr <REMOTE IP>` will be failed. It is the case that you send a `ControlRequest::Start` containing `local_working_dir` args to remote machine, remote machine can't find required source in `local_working_dir`.

Is this true? @haixuanTao

Reply:
connect remote daemon to coordinator may not be the case. How about creating a connection between coordinators?

Reply:
Yes! It is the second point of-> https://github.com/dora-rs/dora/issues/459#issuecomment-2114182167 

But maybe I think we can start by just connecting the CLI, and deal with PATH issues in a second PR.

Reply:
No, design wise, there should only be one coordinator and multiple daemon.

This would be out of scope of this PR.

Reply:
In the example CI, as we´re actually using the same machine, there shouldn´t be PATH issues.